kernel/common/method.rb

```diff
@@ -158,6 +159,23 @@ def unbind
     UnboundMethod.new(@defined_in, @executable, @receiver.class, @name)
   end
 
+  def super_method
+    @ancestors.each_with_index do |a, i|
```

Comment:
As an aside, please don't use one letter variables. I know the codebase is littered with them but it's a huge pain to constantly try and remember what `a` or `x` or another-one-letter-variable stands for.